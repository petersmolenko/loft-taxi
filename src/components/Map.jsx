import React, { useEffect, useRef } from "react";
import { Grid, Paper, TextField, Button } from "@material-ui/core";
import "mapbox-gl/dist/mapbox-gl.css";
import { AuthHOC } from "../AuthContext.jsx";
import PropTypes from "prop-types";
import mapboxgl from "mapbox-gl";
mapboxgl.accessToken =
    "pk.eyJ1IjoicGV0ZXJzbW9sZW5rbyIsImEiOiJjazhydWVoY3UwYnV2M3F0bDV4ZXh1Z2N4In0.Dj8PKEOg2s4sE5YeGGygow";

const Map = ({ stopSubmit }) => {
    const mapContainer = useRef(null);

    useEffect(() => {
            const map = new mapboxgl.Map({
                container: mapContainer.current,
                style: "mapbox://styles/mapbox/navigation-preview-night-v4",
                center: [37.62, 55.75],
                zoom: 12,
            });
            return () => map.remove();
    });

    return (
        <>
            <div
                id="mapBox"
                ref={mapContainer}
                style={{
                    flexGrow: 1,
                }}
            ></div>
            <Grid
                style={{
                    position: "absolute",
                    top: "3rem",
                    left: "3rem",
                    width: "30%",
                    minWidth: "25rem",
                }}
            >
                <Paper elevation={0} className="AppForm">
                    <form noValidate autoComplete="off">
                        <Grid container justify="center" direction="column">
                            <Grid item>
                                <TextField
                                    id="userName"
                                    label="Откуда"
                                    className="AppForm_margin"
                                    fullWidth
                                />
                            </Grid>
                            <Grid item>
                                <TextField
                                    id="userName"
                                    label="Куда"
                                    className="AppForm_margin"
                                    fullWidth
                                />
                            </Grid>
                            <Grid item>
                                <Button
                                    variant="contained"
                                    disabled
                                    fullWidth
                                    onClick={(e) => {
                                        e.preventDefault();
                                        stopSubmit();
                                    }}
                                >
                                    Вызвать такси
                                </Button>
                            </Grid>
                        </Grid>
                    </form>
                </Paper>
            </Grid>
        </>
    );
};

Map.propTypes = {
    stopSubmit: PropTypes.func,
};

export default AuthHOC(Map);
