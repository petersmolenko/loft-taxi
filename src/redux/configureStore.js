import { createStore, applyMiddleware, combineReducers } from 'redux';
import auth, {loginMiddleware, signupMiddleware} from './modules/auth'
import profile, {fetchProfileMiddleware} from './modules/profile'

const createStoreWithMiddleware = applyMiddleware( loginMiddleware, signupMiddleware, fetchProfileMiddleware)(createStore); 

const reducer = combineReducers({
  auth, profile
});

const configureStore = (initialState) => createStoreWithMiddleware(reducer, initialState);
export default configureStore;