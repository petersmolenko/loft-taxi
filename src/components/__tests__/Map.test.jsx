import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Map from "../Map";
import { Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { createBrowserHistory } from "history";

const mockStore = configureStore([]);

jest.mock("mapbox-gl/dist/mapbox-gl", () => ({
    Map: jest.fn(function () {
        this.remove = () => {};
    }),
}));

describe("Map testing", () => {
    it("renders correctly (authorized)", () => {
        const store = mockStore({
            auth: { isLoggedIn: false },
            routes: {
                addresses: ['address1', 'address2']
            },
            profile: {
                paymentInfo: {info: 'info'}
            }
        });
        const history = createBrowserHistory();

        const tree = renderer
            .create(
                <Provider store={store}>
                    <Router history={history}>
                        <Map />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
