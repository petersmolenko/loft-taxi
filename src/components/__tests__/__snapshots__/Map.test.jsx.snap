// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map testing renders correctly (authorized) 1`] = `
Array [
  <div
    id="mapBox"
    style={
      Object {
        "flexGrow": 1,
      }
    }
  />,
  <div
    className="MuiGrid-root"
    style={
      Object {
        "left": "3rem",
        "minWidth": "25rem",
        "position": "absolute",
        "top": "3rem",
        "width": "30%",
      }
    }
  >
    <div
      className="MuiPaper-root AppForm MuiPaper-elevation0 MuiPaper-rounded"
    >
      <form
        autoComplete="off"
        noValidate={true}
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-justify-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="MuiFormControl-root MuiTextField-root AppForm_margin MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink={false}
                htmlFor="userName"
                id="userName-label"
              >
                Откуда
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  id="userName"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="MuiFormControl-root MuiTextField-root AppForm_margin MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink={false}
                htmlFor="userName"
                id="userName-label"
              >
                Куда
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  id="userName"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled MuiButton-fullWidth Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Вызвать такси
              </span>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
]
`;
